thetis/callback.py

```diff
@@ -646,15 +654,22 @@ def _initialize(self):
         if outputdir is None:
             outputdir = self.solver_obj.options.outputdir
 
+        # construct mesh points
+        xyz = (self.x, self.y, self.z) if self.on_sphere else (self.x, self.y)
+        self.xyz = numpy.array([xyz])
+
         # test evaluation
         try:
-            self.solver_obj.fields.bathymetry_2d.at((self.x, self.y))
+            if self.eval_func is None:
+                self.solver_obj.fields.bathymetry_2d.at(self.xyz, tolerance=self.tolerance)
```

Comment:
It would be good to replace the `at` calls in the callbacks with the `VertexOnlyMesh` approach, so that you keep the interpolator (AFAIK `at` doesn't do that). It's on my to do list.

Reply:
Although I don't think `VertexOnlyMesh` currently accepts a tolerance, so we might need that first.

Reply:
Yes, I was planning to do that in another PR. If we can cope without a tolerance that would be idea of course but there might be some issues with points close to the boundary.